commit 8f1146aad251da78d70ffdf686f9e1a6289d6a30
Author: John Spray <jspray@redhat.com>
Date:   Thu Jun 8 08:43:02 2017 -0400

    mgr: don't let TYPE_CLIENT entities send stats
    
    Anyone instantiating librados instances ended up
    appearing as a weird entry in DaemonStateIndex.
    
    Come back to this at the point we want e.g. stats
    from RGWs or NFS gateways.
    
    Signed-off-by: John Spray <john.spray@redhat.com>

diff --git a/src/mgr/DaemonServer.cc b/src/mgr/DaemonServer.cc
index faa60a77659..baef94b5f6f 100644
--- a/src/mgr/DaemonServer.cc
+++ b/src/mgr/DaemonServer.cc
@@ -271,7 +271,12 @@ bool DaemonServer::handle_open(MMgrOpen *m)
           << m->daemon_name << dendl;
 
   auto configure = new MMgrConfigure();
-  configure->stats_period = g_conf->mgr_stats_period;
+  if (m->get_connection()->get_peer_type() == entity_name_t::TYPE_CLIENT) {
+    // We don't want clients to send us stats
+    configure->stats_period = 0;
+  } else {
+    configure->stats_period = g_conf->mgr_stats_period;
+  }
   m->get_connection()->send_message(configure);
 
   if (daemon_state.exists(key)) {
@@ -292,6 +297,13 @@ bool DaemonServer::handle_report(MMgrReport *m)
   dout(4) << "from " << m->get_connection() << " name "
           << m->daemon_name << dendl;
 
+  if (m->get_connection()->get_peer_type() == entity_name_t::TYPE_CLIENT) {
+    // Clients should not be sending us stats
+    dout(4) << "rejecting report from client " << m->daemon_name << dendl;
+    m->put();
+    return true;
+  }
+
   DaemonStatePtr daemon;
   if (daemon_state.exists(key)) {
     dout(20) << "updating existing DaemonState for " << m->daemon_name << dendl;
